puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/lib/rpm_packaging.groovy

```diff
@@ -17,3 +17,67 @@ def find_changed_packages(diff_range) {
 
     return changed_packages.split()
 }
+
+def query_rpmspec(specfile, queryformat) {
+    result = sh(returnStdout: true, script: "rpmspec -q --srpm --undefine=dist --undefine=foremandist --queryformat=${queryformat} ${specfile}").trim()
+    return result
+}
+
+    stage("Verify version and release"){
+        if (build_type == 'scratch') {
+            def packages = packages_to_build.split(' ')
+
+            for (int i = 0; i < packages.size(); i++) {
+                package_name = packages[i]
+                version_status = 'failed'
+                release_status = 'failed'
+
+                new_version = query_rpmspec("packages/${package_name}/*.spec", '%{VERSION}')
+                new_release = query_rpmspec("packages/${package_name}/*.spec", '%{RELEASE}')
+
+                sh "git checkout origin/${env.gitlabTargetBranch}"
+                if (fileExists("packages/${package_name}")) {
+                    old_version = query_rpmspec("packages/${package_name}/*.spec", '%{VERSION}')
+                    old_release = query_rpmspec("packages/${package_name}/*.spec", '%{RELEASE}')
+
+                    sh "git checkout -"
+
+                    compare_version = sh(
+                      script: "rpmdev-vercmp ${old_version} ${new_version}",
+                      returnStatus: true
+                    )
+
+                    compare_release = sh(
+                      script: "rpmdev-vercmp ${old_release} ${new_release}",
+                      returnStatus: true
+                    )
+
+                    compare_new_to_one = sh(
+                      script: "rpmdev-vercmp 1 ${new_release}",
+                      returnStatus: true
+                    )
+
+                    if (compare_version != VERCMP_EQUAL && (compare_new_to_one == VERCMP_OLDER || compare_new_to_one == VERCMP_EQUAL)) {
```

Comment:
Edge case, but what if the version actually goes backwards? This can happen if a PR was merged in the mean time and an incorrect rebase.

Reply:
Agreed, edge case not sure worth taking into account.